detekt-core/src/main/kotlin/io/gitlab/arturbosch/detekt/core/KtTreeCompiler.kt

```
@@ -57,12 +58,43 @@ class KtTreeCompiler(
 
     private fun Path.isKotlinFile() = extension in KT_ENDINGS
 
-    private fun isIgnored(path: Path): Boolean {
-        val ignored = pathFilters?.isIgnored(path)
-        if (ignored == true) {
-            settings.debug { "Ignoring file '$path'" }
+    private inner class CollectingFileVisitor : FileVisitor<Path> {
+        private val _collected: MutableList<Path> = mutableListOf()
+        val collected: List<Path>
+            get() = _collected
+
+        override fun preVisitDirectory(dir: Path, attrs: BasicFileAttributes): FileVisitResult {
+            if (dir.isIgnored()) {
+                settings.debug { "Ignoring subtree '$dir'." }
+                return FileVisitResult.SKIP_SUBTREE
+            }
+            return FileVisitResult.CONTINUE
+        }
+
+        override fun visitFile(file: Path, attrs: BasicFileAttributes): FileVisitResult {
+            if (!file.isKotlinFile()) {
+                return FileVisitResult.CONTINUE
+            }
+            if (file.isIgnored()) {
+                settings.debug { "Ignoring file '$file'" }
+            } else {
+                _collected.add(file)
+            }
+            return FileVisitResult.CONTINUE
+        }
+
+        override fun visitFileFailed(file: Path, exc: IOException): FileVisitResult {
+            settings.error("Error visiting file '$file'.", exc)
+            return FileVisitResult.TERMINATE
```

Should we just throw the exception? If we have an exception at this point the issue is really big and if we just ignore the scan will be only partial.

I know that this is a common pattern on detekt: Try to no fail. But this make the work of debugging issues way harder.

---

That is a good point. I have changed this.